### PR TITLE
Replace dependency on sasl with pure-sasl

### PR DESCRIPTION
**Pull request by nonsleepr**
Installation of python-sasl is complicated on Windows, while pure-sasl is pure.

### CODE REVIEW

**Comment by laserson** [comment]
Thanks for this!  I'm a fan of things that make it easier for people to use.  However, AFAIK the rest of the Hadoop stack uses the `python-sasl` package, which I am wary of mucking with.  My preference would be to have a `sasl_compat.py` module which defaults to `python-sasl` and falls back on `pure-sasl` while emitting a warning.  You can then abstract over the two modules.

cc @caseyching @ishaan

File: impala/tests/test_dbapi_connect.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Copyright 2013 Cloudera Inc.
```
Comment (laserson) [inline impala/tests/test_dbapi_connect.py]: These tests look great!  Though I need to see if I have access to internal test clusters for all the options.

File: impala/thrift_sasl.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 import sys
 import struct
 
-import sasl
+#import sasl
```
Comment (laserson) [inline impala/thrift_sasl.py]: Just delete this if it's unused.

**Comment by laserson** [comment]
@nonsleepr Just to check, are you going to resubmit a modified PR?

File: impala/tests/test_dbapi_connect.py (hunk 1 above)
Comment (nonsleepr) [inline impala/tests/test_dbapi_connect.py]: Only 3 of 8 are successful though with default Cloudera VM settings.

**Comment by nonsleepr** [comment]
@laserson Resubmitted as #91.

**Pull request closed without merging**